src/tools/metrics.py

```diff
@@ -0,0 +1,188 @@
+import uuid
+
+from api.models import AuthUser, Project
+
+
+SUMMARY_URL_SUFFIXES = (
+    "/beltfishes/obstransectbeltfishes/",
+    "/beltfishes/sampleunits/",
+    "/beltfishes/sampleevents/",
+    "/benthiclits/obstransectbenthiclits/",
+    "/benthiclits/sampleunits/",
+    "/benthiclits/sampleevents/",
+    "/benthicpits/obstransectbenthicpits/",
+    "/benthicpits/sampleunits/",
+    "/benthicpits/sampleevents/",
+    "/bleachingqcs/obscoloniesbleacheds/",
+    "/bleachingqcs/obsquadratbenthicpercents/",
+    "/bleachingqcs/sampleunits/",
+    "/bleachingqcs/sampleevents/",
+    "/habitatcomplexities/obshabitatcomplexities/",
+    "/habitatcomplexities/sampleunits/",
+    "/habitatcomplexities/sampleevents/",
+    "/benthicpqts/obstransectbenthicpqts/",
+    "/benthicpqts/sampleunits/",
+    "/benthicpqts/sampleevents/",
+    # "/summarysampleevents/",
+)
+SUMBIT_URL_SUFFIX = "/submit/"
+PROJECT_URL_PREFIX = "/v1/projects/"
+SUMMARY_EVENT_TYPE = "summary"
+PROJECT_EVENT_TYPE = "project"
+SUBMIT_EVENT_TYPE = "submit"
+OTHER_EVENT_TYPE = "other"
+EVENT_TYPES_FILTER = (SUMMARY_EVENT_TYPE, PROJECT_EVENT_TYPE, SUBMIT_EVENT_TYPE)
+PROFILE_DEFAULT = {"first_name": "", "last_name": "", "email": ""}
+PROJECT_DEFAULT = {"project_name": "", "project_tags": "", "profiles": {}}
+
+
+def meta(method):
+    def _meta(*args, **kw):
+        args = list(args)
+        log_event = args[0]
+        if hasattr(log_event, "meta"):
```

Comment:
can't this if/else be replaced by...

`log_event.meta = log_event.meta or {}`

Reply:
As an obj I don't think that would quite do it but see what I did try to clean it up.